lib/features/wallet_screen/wallet_screen.dart

```diff
@@ -0,0 +1,83 @@
+part of '../feature.dart';
+
+class WalletScreen extends ConsumerWidget {
+  const WalletScreen({super.key});
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final isMobile = ScreenLayout.isMobile(context);
+    final isPortrait =
+        MediaQuery.of(context).orientation == Orientation.portrait;
```

Comment:
add this to `build_context_extension`

bool get isPortrait =>
      MediaQuery.of(this).orientation == Orientation.portrait;